packages/cli/src/commands/trace.ts

```diff
@@ -45,6 +45,17 @@ export async function trace({
   // will call `trace_transaction`, and decode as much data from the trace
   // as possible, the same way that an error occurs
 
+  if (providerUrl) {
+    // get chain id from provider
+    const publicClient = viem.createPublicClient({
+      transport: viem.http(providerUrl),
+    });
+
+    if (chainId !== await publicClient.getChainId()) {
+      throw new Error("The provided chain ID does not match the provider url's chain ID, please ensure both chain ID's match or specify only one chain ID.")
```

Comment:
Added error for mismatching chain id

Reply:
![image](https://github.com/usecannon/cannon/assets/30664234/791df4b9-8f23-4875-a80d-1d3ccf8fea94)

Reply:
Removed required from chain id option and added a custom error when provider url and chain id are undefined:
![image](https://github.com/usecannon/cannon/assets/30664234/313d0279-7603-43ef-8314-df2455b70092)

Also edited the error message on chain id mismatch:
![image](https://github.com/usecannon/cannon/assets/30664234/a915d3d3-7eaa-444d-8c9b-2e5e25270be0)